### PR TITLE
fix(ingest/delta-lake): fix CI issues due to delta lake version bump

### PR DESCRIPTION
**Pull request by mayurinehate**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py [hunk 1]
```diff
@@ -159,7 +159,7 @@ def _create_operation_aspect_wu(
             reported_time: int = int(time.time() * 1000)
             last_updated_timestamp: int = hist["timestamp"]
             statement_type = OPERATION_STATEMENT_TYPES.get(
-                hist.get("operation"), OperationTypeClass.CUSTOM
+                hist.get("operation", "UNKNOWN"), OperationTypeClass.CUSTOM
```
[the remark this answers is not in the document]
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py:162]: This is to get rid of mypy error at this line. :(

**Review by jjoyce0510 — COMMENTED** [review]
Thanks for handling this!

File: metadata-ingestion/src/datahub/ingestion/source/delta_lake/delta_lake_utils.py [hunk 2]
```diff
@@ -1,7 +1,20 @@
...
+
+try:
+    from deltalake.exceptions import TableNotFoundError
+
+    DELTALAKE_VERSION_GTE_0_10_0 = True
+
+
+except ImportError:
+    # For deltalake < 0.10.0
+    from deltalake import PyDeltaTableError  # type: ignore[attr-defined]
```
[the remark this answers is not in the document]
Comment (asikowitz) [reply metadata-ingestion/src/datahub/ingestion/source/delta_lake/delta_lake_utils.py:14]: Previous changes looked more like this but there were mypy issues. Not exactly sure of the situation though

[the remark this answers is not in the document]
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/delta_lake/delta_lake_utils.py:14]: yeah, it gave "no-redef" mypy error.

**Pull request merged**